src/DemaConsulting.TestResults/IO/JUnitSerializer.cs

```diff
@@ -129,6 +74,125 @@ public static string Serialize(TestResults results)
+
+        // Add test cases
+        foreach (var test in suiteTests)
+        {
+            var testCase = CreateTestCaseElement(test);
+            testSuite.Add(testCase);
+        }
+
+        return testSuite;
+    }
+
+    /// <summary>
+    ///     Determines if an outcome represents an error condition
+    /// </summary>
+    /// <param name="outcome">The test outcome to evaluate</param>
+    /// <returns>True if the outcome is Error, Timeout, or Aborted; otherwise false</returns>
+    private static bool IsErrorOutcome(TestOutcome outcome)
+    {
+        return outcome == TestOutcome.Error || outcome == TestOutcome.Timeout || outcome == TestOutcome.Aborted;
+    }
+
+    /// <summary>
+    ///     Creates a test case element
+    /// </summary>
+    /// <param name="test">The test result to serialize</param>
+    /// <returns>An XElement representing a testcase with outcome, system output, and system error</returns>
+    private static XElement CreateTestCaseElement(TestResult test)
+    {
+        var testCase = new XElement("testcase",
+            new XAttribute("name", test.Name),
+            new XAttribute("classname", string.IsNullOrEmpty(test.ClassName) ? DefaultSuiteName : test.ClassName),
+            new XAttribute("time", test.Duration.TotalSeconds.ToString(TimeFormatString, CultureInfo.InvariantCulture)));
+
+        // Add failure or error information based on outcome
+        AddOutcomeElement(testCase, test);
+
+        // Add system output
+        if (!string.IsNullOrEmpty(test.SystemOutput))
+        {
+            testCase.Add(new XElement("system-out", new XCData(test.SystemOutput)));
+        }
+
+        // Add system error
+        if (!string.IsNullOrEmpty(test.SystemError))
+        {
+            testCase.Add(new XElement("system-err", new XCData(test.SystemError)));
+        }
+
+        return testCase;
+    }
+
+    /// <summary>
+    ///     Adds the appropriate outcome element (failure, error, or skipped) to a test case
+    /// </summary>
+    /// <param name="testCase">The testcase element to add the outcome element to</param>
+    /// <param name="test">The test result containing the outcome information</param>
+    private static void AddOutcomeElement(XElement testCase, TestResult test)
+    {
+        if (test.Outcome == TestOutcome.Failed)
+        {
+            var failure = CreateFailureOrErrorElement("failure", test);
+            testCase.Add(failure);
+        }
+        else if (IsErrorOutcome(test.Outcome))
+        {
+            var error = CreateFailureOrErrorElement("error", test);
+            testCase.Add(error);
+        }
+        else if (!test.Outcome.IsExecuted())
+        {
+            var skipped = new XElement("skipped");
+            if (!string.IsNullOrEmpty(test.ErrorMessage))
+            {
+                skipped.Add(new XAttribute("message", test.ErrorMessage));
```

Comment:
The "message" literal here is triggering a SonarQube complaint: "Define a constant instead of using this literal 'message' 5 times."

Please fix this complaint.